editors/code/rust.tmGrammar.json

```diff
@@ -180,7 +180,7 @@
             "begin": "\\b(extern)\\s+(crate)",
             "beginCaptures": {
                 "1": {
-                    "name": "keyword.control.rust"
+                    "name": "storage.type.rust"
```

Comment:
I've just realized. This might be wrong. I think this should've been `keyword.other.rust`. The `extern` keyword doesn't signify any storage here.

Reply:
I have never been clear on precisely what constitutes `storage` in Textmate grammars. The official documentation states that `storage` should be used for 'things relating to "storage,'" which is a little bit circular for the uninitiated. @bjorn3 had specific opinions when I originally proposed the grammar, so perhaps could weigh in here. My impression is that `storage` applies to situations where memory is allocated, but there is a lot of gray area there. CC: @georgewfraser.

I'm happy to merge a reverting PR on these two scopes, but let's hold off for some consensus. As an aside, the grammar probably needs tests so that it doesn't become a brittle house of cards when others contribute. I'll have a look at implementing [PanAeon's approach](https://github.com/PanAeon/vscode-tmgrammar-test) when things slow down on the issue front.

Reply:
I think this should be `storage.modifier`. See https://macromates.com/manual/en/language_grammars